### ISSUE TITLE
Live preview for campaign templates not working on multi site.

### ISSUE THREAD

**Issue by watarutmnh** [putyourlightson/craft-campaign#251]
### Describe the bug
While I editing some campaign associated to non-primary site, live preview doesn't work.

### To reproduce
Steps to reproduce the behaviour:
1. Create campaign type and associate with a site which's domain doesn't match with the control panel URL.
2. Create or edit a campaign and enable live preview mode.
3. See console error message:

`Response to preflight request doesn't pass access control check: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.`

### Versions
- Plugin version: 1.20.2
- Craft version: 3.7.10

**Comment by bencroker** [comment]
As you've already discovered in https://github.com/craftcms/cms/issues/7851, this is an issue with Craft. Campaign will be moving to the new Preview system in version 2.

**Comment by watarutmnh** [comment]
@bencroker Okay cool. Thank you for taking care this issue.
I will wait version 2.

**Comment by bencroker** [comment]
I've since opted to add it as per https://github.com/craftcms/cms/issues/7851#issuecomment-904865027

Can you please check if this solves the issue for you?

**Comment by watarutmnh** [comment]
@bencroker I replaced my src/controllers/CampaignsController.php with your updated one:
https://github.com/putyourlightson/craft-campaign/commit/656e57fc19eb0bf015268ee59ca8428eb5c217d6
But it still errors...

**Comment by bencroker** [comment]
What is `Access-Control-Allow-Origin` being set to in the network response to the live preview request?

**Comment by watarutmnh** [comment]
@bencroker May I ask how to check value of `Access-Control-Allow-Origin` ? I'm not familiar with this stuff.. 
I attached the console error.
![screenshot 2021-08-26 9 35 48](https://user-images.githubusercontent.com/6081378/130882427-1b819b97-ce7b-45df-b533-b61d5303e1ea.png)

**Comment by bencroker** [comment]
Look under _Response Headers_ in the _Network_ tab of your browser console (Chrome screenshot below).

<img width="597" alt="Screenshot 2021-08-26 at 10 11 02" src="https://user-images.githubusercontent.com/57572400/130926419-1851bf57-4f63-4447-850c-068aa44a43bf.png">

**Comment by watarutmnh** [comment]
@bencroker It's strange that I can not find the `Access-Control-Allow-Origin` .
![screenshot 2021-08-27 15 05 50](https://user-images.githubusercontent.com/6081378/131080361-18f471fb-2202-44ae-b655-9e3e432dd7aa.png)

Am I doing something wrong?

**Comment by bencroker** [comment]
You need to look at the live preview XHR request.

**Comment by bencroker** [comment]
@watarutmnh Had a chance to test yet?

**Comment by watarutmnh** [comment]
@bencroker  Sorry for delay. I can not find the `Access-Control-Allow-Origin`.
![screenshot 2021-08-31 at 16 12 35](https://user-images.githubusercontent.com/6081378/131459219-0aae9c3e-0cd2-4115-be29-500694db7269.png)

**Comment by bencroker** [comment]
Please ensure you are using the `develop` branch in composer to test.

```
composer require putyourlightson/craft-campaign:dev-develop
```

**Comment by watarutmnh** [comment]
Okay, I changed it to `develop` branch, but the result is the same.
![screenshot 2021-09-01 at 15 04 46](https://user-images.githubusercontent.com/6081378/131620380-b0c0b9e7-0a56-4370-b4e2-f76de1e144b6.png)

**Comment by watarutmnh** [comment]
I have a CORS error. Does it be any help?
![screenshot 2021-09-01 at 15 03 23](https://user-images.githubusercontent.com/6081378/131620469-98fdaa41-4c60-4fb1-892b-fa9074b34c81.png)

**Comment by bencroker** [comment]
You need to inspect the live preview request itself, which is sent when you change one of the campaign fields when the live preview window is open. Ensure you are looking at `Fetch/XHR` requests and the URL should contain `?token=...`.

**Comment by watarutmnh** [comment]
I think I'm doing as you explained, but how do you think?
![screenshot 2021-09-02 at 9 47 31](https://user-images.githubusercontent.com/6081378/131764131-374e9ddf-2675-49a8-aff6-74ebab75a57d.png)

**Comment by watarutmnh** [comment]
Also the CORS error says "Preflight wildcard origin not allowed".
![screenshot 2021-09-02 at 9 54 27](https://user-images.githubusercontent.com/6081378/131764813-cdafa9fe-5526-4a27-ab9d-558ad6a6f7d1.png)

**Comment by bencroker** [comment]
I took a fresh look at this and adding the following header to your secondary site should work.

```
Access-Control-Allow-Origin: https://primary-site.com
```

You can do this in one of several ways, there are plenty of online resources, or use the [Sherlock plugin](https://putyourlightson.com/plugins/sherlock). Let me know how that goes for you.

**Comment by bencroker** [comment]
Have you tested this approach @watarutmnh?

**Comment by bencroker** [comment]
Ok so I've opted for a different approach, adding a new `allowedOrigins` config setting in https://github.com/putyourlightson/craft-campaign/commit/906d68a53028a6cd2f2d08e1acb8705575ddb4f6

Now it should be a simple matter of adding your primary domain to the config file:
https://github.com/putyourlightson/craft-campaign/blob/906d68a53028a6cd2f2d08e1acb8705575ddb4f6/src/config.php#L39-L42

Let me know how that goes for you @watarutmnh.

**Comment by bencroker** [comment]
Released in version 1.21.0.
